unicore/distribute/utils.py

```diff
@@ -299,3 +322,82 @@ def get_config(request):  # pragma: no cover
         The HTTP request
     """
     return request.registry.settings
+
+
+def get_schema_names(repo):
+    schema_files = glob.glob(
+        os.path.join(repo.working_dir, '_schemas', '*.avsc'))
+    names = []
+    for schema_file in schema_files:
+        names.append(
+            schema_file[schema_file.rfind("/") + 1:schema_file.rfind(".")])
+    return names
+
+
+def add_model_item_to_pull_dict(storage_manager, path, pull_dict):
+    if path.endswith(".json"):
+        model = storage_manager.load(path)
+        pull_dict[model.__module__ + "." +
+                  model.__class__.__name__].append(dict(model))
+        return True
+    return False
+
+
+def get_repository_diff(repo, commit_id):
```

Comment:
Thanks. Got help understanding what to do when multiple yields are present. changed it to use format_diffindex